### ISSUE TITLE
Code Push app update does not apply in RN 0.57.1

### ISSUE THREAD

**Issue by sundeepsingh12** [Microsoft/react-native-code-push#1420]
### Steps to Reproduce
1. Install lower version react-native app
2. Release code push higher deployment
3. Check for the version, and call for code push update.

**Code** for implementing code push update :

codePush.sync({
      updateDialog: true,
      installMode: codePush.InstallMode.IMMEDIATE,
      deploymentKey: androidDeploymentKey,
    }, (status) => {
      switch (status) {
        case codePush.SyncStatus.CHECKING_FOR_UPDATE:
         break;
        case codePush.SyncStatus.DOWNLOADING_PACKAGE:
          break;
        case codePush.SyncStatus.INSTALLING_UPDATE:
         break;
        case codePush.SyncStatus.UP_TO_DATE:
         break;
        case codePush.SyncStatus.UPDATE_INSTALLED:
         break;
        case codePush.SyncStatus.AWAITING_USER_ACTION:
         break;
        case codePush.SyncStatus.SYNC_IN_PROGRESS:
         break;
        default:
      }

### Expected Behavior
After code push install complete, the application should have updated code and version should upgrade.

### Actual Behavior
Once installation is complete, the new changes are not reflected in the app, the code-push debug error is as shown below.

![image](https://user-images.githubusercontent.com/21218662/46204182-f1ad0380-c339-11e8-8866-d88899ecba73.png)

### Reproducible Demo

NA

### Environment

* react-native-code-push version: 5.4.2

* react-native version: **0.57.1**

* iOS/Android/Windows version: Android 7.0 (Nougat)

* Does this reproduce on a debug build or release build?
   : It is reproduced on release build.

* Does this reproduce on a simulator, or only on a physical device?
   : We are facing this on Physical device

**Comment by alexandergoncharov** [comment]
Hi @sundeepsingh12 ,
Thanks for reporting!

I just retested react-native@0.57.1 version and haven't found any errors. This issue appears in case if CodePush can't set JsBundle. So, possible there can be other issue.
Could you please provide dummy project with detailed reprosteps? 

Thanks,
Alexander

**Comment by alexandergoncharov** [comment]
Also, could you please clarify what exactly version was used for creating app and releasing CodePush update?

**Comment by sundeepsingh12** [comment]
Hi @alexandergoncharov 
Thanks for your timely reply!

Please find the link for demo project in which we are still facing the issue : 
https://github.com/sundeepsingh12/SampleCodePush.git

About version we used : 
react-native : 0.57.1
react-native-code-push : 5.4.2

for both creating as well as releasing code push.

Please have look at this and suggest what could we possibly do to resolve this problem.

**Comment by alexandergoncharov** [comment]
Hi @sundeepsingh12 ,

Yeah, I reproduced issue with release build. Investigating.

**Comment by NickToropov** [comment]
Hey guys! The issue is in minifying. It's required to keep `mBundleLoader` field of `ReactInstanceManager` unminified.

Setup your proguard file with the following rule:
```
-keepclassmembers class com.facebook.react.ReactInstanceManager {
    private final ** mBundleLoader;
}
```
Please, try this fix and let me know  if it solves the issue.

**Comment by NickToropov** [comment]
Created a [PR#1425](https://github.com/Microsoft/react-native-code-push/pull/1425)

**Comment by ivanbtrujillo** [comment]
Hi @NickToropov, I've updated the proguard-rules.pro with your changes but It fails. Once the app detect the update and tries to refresh, it crash.

My config is quite simple (the default codepush config), and it was working before update to v0.57.

**Comment by ivanbtrujillo** [comment]
I've installed from master branch after your changes were merged but still doesn't work
```
yarn add https://github.com/Microsoft/react-native-code-push.git
```

**Comment by NickToropov** [comment]
@ivanbtrujillo what error do you get? Could you share output from logcat with error related to this issue?

**Comment by NickToropov** [comment]
Please take a look at attached demo: 
[Issue1420.zip](https://github.com/Microsoft/react-native-code-push/files/2456428/Issue1420.zip)
execute `yarn` command in the root of the project, insert your deployment key in `Issue1420/android/app/src/main/res/values/strings.xml`, update singing config `Issue1420/android/app/build.gradle`, switch build variant to Release and run the app. Then publish CodePush updates and observe app behavior.

**Comment by ivanbtrujillo** [comment]
Hi @NickToropov I've used your demo and it works, but in my project not. The error I've received from logcat is the following:

```
10-09 09:42:19.405 16857 16919 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
10-09 09:42:19.405 16857 16919 E AndroidRuntime: Process: com.blueskytechnologyeu.folloow, PID: 16857
10-09 09:42:19.405 16857 16919 E AndroidRuntime: java.lang.IllegalArgumentException: Unable to find JSIModule for class interface com.facebook.react.bridge.UIManager
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.JSIModuleRegistry.getModule(JSIModuleRegistry.java:17)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.CatalystInstanceImpl.getJSIModule(CatalystInstanceImpl.java:478)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.uimanager.UIManagerHelper.getUIManager(UIManagerHelper.java:30)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.uimanager.UIManagerModule.dispatchViewManagerCommand(UIManagerModule.java:600)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:761)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:98)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:156)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192)
10-09 09:42:19.405 16857 16919 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:776)
```

There is another issue with the same problem:
[https://github.com/facebook/react-native/issues/20636#issuecomment-425248856](url)

**Comment by ivanbtrujillo** [comment]
Yep that was the issue. It's working.

**Comment by NickToropov** [comment]
@ivanbtrujillo I'm glad to hear your issue was resolved.
So I'll close this issue for now as it seems to be resolved.
Please feel free to reopen it you have any question or create a new one.

**Comment by sundeepsingh12** [comment]
@alexandergoncharov @NickToropov We are still facing the issue.

Though it is different this time, the app get updated after the code push, but as soon as we destroy the app/clear from recents it agains revert back to older version.

Please suggest !

**Comment by Udbhav12** [comment]
@NickToropov - The changes you suggested is android specific. This problem occurs in ios also

**Comment by alexandergoncharov** [comment]
@sundeepsingh12 ,

Could you please provide logs during update, repro steps and some dummy project with reproducing issue?

**Comment by NickToropov** [comment]
@sundeepsingh12, @Udbhav12 we've fixed issue mentioned first(`Unable to set JsBundle...`). Now I'm sure you've faced with another one so please provide us with detailed steps or demo project that we could reproduce your new issue and try to fix it.

**Comment by Udbhav12** [comment]
@NickToropov Please find the code-push debug logs for android when app is destroyed

As you can see from the logs that code push update is successfully applied initially, but as soon as the app is destroyed and then the app is reopened, the update applied by code push is rolled back.
Btw (Unable to set JsBundle...) has been fixed by the change you suggested

![screen shot 2018-10-11 at 5 35 37 pm](https://user-images.githubusercontent.com/16081461/46802984-a1439600-cd7c-11e8-8644-364c92a3a488.png)

**Comment by NickToropov** [comment]
@Udbhav12 the issue is that you don't call `notifyAppReady`
Please check in this docs:
1. https://github.com/Microsoft/react-native-code-push#debugging--troubleshooting 
```"If you're not calling sync on app start (e.g. within componentDidMount of your root component), then you need to explicitly call notifyApplicationReady on app start, otherwise, the plugin will think your update failed and roll it back."```
1. https://github.com/Microsoft/react-native-code-push/blob/master/docs/api-js.md#codepushnotifyappready
